modules/core/src/alloy/proto/validation/ProtoIndexTraitValidator.java

```diff
@@ -90,18 +94,54 @@ private List<ValidationEvent> validateShape(Model model, Shape shape) {
 					.collect(Collectors.groupingBy(e -> e.getValue().getNumber(),
 							Collectors.mapping(Map.Entry::getKey, Collectors.toList())));
 
-			return perIndex.entrySet().stream().filter(e -> e.getValue().size() > 1).map(e -> {
+			indexCheck = perIndex.entrySet().stream().filter(e -> e.getValue().size() > 1).map(e -> {
 				final Integer index = e.getKey();
 				final List<String> labels = e.getValue();
 				final String memberNames = labels.stream().collect(Collectors.joining(","));
 				return ValidationEvent
 						.builder().id(DUPLICATED_PROTO_INDEX).message("Proto index " + index
-								+ " is used muliple times in members " + memberNames + " of shape " + shape + ".")
+								+ " is used multiple times in members " + memberNames + " of shape " + shape + ".")
 						.shape(shape).severity(Severity.ERROR).build();
-			}).collect(Collectors.toList());
+			});
 		} else {
-			return Collections.emptyList();
+			indexCheck = Stream.empty();
 		}
+
+		Stream<ValidationEvent> openStringEnumChecks = null;
+		if ((shape.isEnumShape() || shape.isIntEnumShape()) && shape.hasTrait(OpenEnumTrait.class)) {
+			if (noFieldIsIndexed) {
+				openStringEnumChecks = Stream.empty();
+			} else {
+				openStringEnumChecks = Stream.of(ValidationEvent.builder().id(OPEN_ENUM_MUST_NOT_HAVE_INDEXES)
+						.message("Members of enumeration" + shape + "must not carry proto indexes.").shape(shape)
+						.severity(Severity.ERROR).build());
+			}
+		} else {
+			Stream.empty();
+		}
+
+		Stream<ValidationEvent> enumHasZeroChecks = null;
+
+		if ((shape.isEnumShape() || shape.isIntEnumShape()) && !shape.hasTrait(OpenEnumTrait.class)
+				&& allFieldsAreIndexed) {
+			if (fieldsAndIndexes.containsValue(Optional.of(new ProtoIndexTrait(0)))) {
+				enumHasZeroChecks = Stream.empty();
+			} else {
+				System.out.println(allFieldsAreIndexed);
+				fieldsAndIndexes.entrySet().forEach(System.out::println);
+				ValidationEvent event = ValidationEvent.builder().id(ENUM_MUST_HAVE_ZERO)
+						.message("One of the members of enumeration" + shape
+								+ "must represent a default value, with proto index set to 0.")
```

Comment:
`protoIndex` maybe even `@protoIndex` ?

Reply:
unless proto index is a valid protobuf term (but i think it's only one we made up because that
s how we called the annotation)

Reply:
[28f904e](https://github.com/disneystreaming/alloy/pull/129/commits/28f904e489467a6b9afa4256d5bfa5469ad8c5d8)